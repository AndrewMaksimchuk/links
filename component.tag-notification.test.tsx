import { describe, test, expect } from 'bun:test'
import { renderToReadableStream } from 'hono/jsx/streaming'
import { TagNotification } from './component.tag-notification';


describe('Component TagNotification', () => {
    test('without props', async () => {
        const stream = renderToReadableStream(<TagNotification />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('with name, color and tagId', async () => {
        const stream = renderToReadableStream(<TagNotification name='web' color='blue' tagId={0} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('without delete button', async () => {
        const stream = renderToReadableStream(<TagNotification text='fish text' />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })
})
