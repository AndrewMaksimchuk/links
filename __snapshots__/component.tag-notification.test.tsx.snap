// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TagNotification without props 1`] = `"<span style="color:initial;background-color:initial30;;border:solid 1px initial;border-radius:1rem;padding:0.125rem 0.5rem 0.25rem;display:flex;align-items:center"> <button type="button" class="outline" style="color:initial;border-color:initial;padding:0;;border-radius:100%;;width:3ch;;aspect-ratio:1;;font-size:0.75rem;;margin:0;;margin-left:0.5rem;" name="tagId" value="NaN" hx-post="/tag-delete" hx-swap="delete" hx-target="closest span">X</button></span>"`;

exports[`Component TagNotification with name, color and tagId 1`] = `"<span style="color:blue;background-color:blue30;;border:solid 1px blue;border-radius:1rem;padding:0.125rem 0.5rem 0.25rem;display:flex;align-items:center">web <button type="button" class="outline" style="color:blue;border-color:blue;padding:0;;border-radius:100%;;width:3ch;;aspect-ratio:1;;font-size:0.75rem;;margin:0;;margin-left:0.5rem;" name="tagId" value="0" hx-post="/tag-delete" hx-swap="delete" hx-target="closest span">X</button></span>"`;

exports[`Component TagNotification without delete button 1`] = `"<span style="color:initial;background-color:initial30;;border:solid 1px initial;border-radius:1rem;padding:0.125rem 0.5rem 0.25rem;display:flex;align-items:center"> fish text</span>"`;
